### ISSUE TITLE
Coordinate system of .trc

### ISSUE THREAD

**Issue by JoanCharmant** [davidpagnon/Sports2D#16]
This looks like an awesome project. I tried it on a few videos and it seems pretty robust

I would like to integrate the output in Kinovea. I'll write a parser for the .trc files it produces. 
Could you explain the coordinate system used? I see the .trc says "units" is meters but I'm not sure about where to find the scaling factor in order to remap that to the image size of the video.

**Comment by JoanCharmant** [comment]
Ah, I see this comment 

```
# Select only the keypoints that are in the model from skeletons.py, invert Y axis, divide pixel values by 1000
```

**Comment by JoanCharmant** [comment]
OK, it's working. I made a quick experiment with keeping all the posture visible at once combined with camera motion estimation. 😀

![t6-onion-skinning](https://github.com/user-attachments/assets/ebc275eb-fbc3-4ab2-9865-bac315c88282)

At the moment I'm using these options:

```
sports2d --video_input "long_jump.mp4" --filter False --interp_gap_smaller_than 500 --show_graphs False  --display_angle_values_on None
```

I remove the filtering to get the raw data to match the image better, as I'm going to do the filtering on my side afterwards anyway, and this is more for visual feedback than measurements I think for now. Interpolation seems always better than filling gaps with the last value when there is motion.

I was using `display_angle_values_on=None` to evaluate how well the stick figure sticks to the person, I think it could be good to have an option to not draw the angle shapes either and just have the body model.

**Comment by davidpagnon** [comment]
Hi, sorry I was away from my computer for a few days. I'm glad you figured it out!

I plan to add a feature for calibrating the camera (i.e. to obtain coordinates in meters rather than in px), it will come when I have a moment.

> I think it could be good to have an option to not draw the angle shapes either and just have the body model.
Noted! I don't have time for it right now but I'll include this feature as soon as I get back to work on the toolbox :)

**Comment by davidpagnon** [comment]
I've just realized that you are the author of Kinovea, which is one of the most widely used tools in biomechanics courses around the world. Congrats on making such a tool free and open-source!

Could you keep me updated when you manage to integrate the output with Kinovea? I'll direct people to Kinovea in the doc.

**Comment by JoanCharmant** [comment]
Yes for sure! 

I don't know about .trc format but if they have a provision for exporting raw pixel coordinates it could be interesting as an option. Or document the transform in the readme near the angle conventions.

This issue can be closed as far as I'm concerned. Thanks!

**Comment by davidpagnon** [comment]
Hi, I just released a new version with the option not to display and calculate angles (`--calculate_angles False`)

For your interest, in the next few days (hopefully), I'm going to add some calibration option for computing coordinates in meters rather than in pixels. And then if people do not choose this option, they will get raw pixel coordinates rather than this weird *1000 multiplication.

**Comment by Niels-Engedal** [comment]
Hello!

Before I show my complete lack of knowledge, I just want to say thank you for developing such an awesome tool as Sports2d! 

My question probably arises from a lack of understanding of motion capture, but I couldn't get the answer from the above comments in this issue, so here is my very beginner question(s):

- How are the coordinates structured? Is origin (0,0) not in the bottom left corner?
- Is it possible to scale the coordinates to a more understandable metric, such as pixels in a resolution e.g. 1920x1080p purely through python? I did not understand how to do that with the comment about inverting Y and dividing by 1000.

It might help with a little context:
I am trying to track a gymnast doing a backwards salto, but the trc-coordinates for e.g. "Hip_Y" are quite hard to understand (e.g. -0.583096 at frame 0 and then ~ -0.2848 at max "height"). 

I would really appreciate if there was an "easy" way to convert this to pure pixels or something, but maybe I just know too little to understand why this is difficult to do? (Just to ensure I am clear enough; I would like to be able to plot e.g. the Trajectory of the Hip in both X and Y as I would see it on the 1920x1080p video. So a starting position could maybe be (X: 960, Y: 360) and then when they jump the Y would increase with the appropriate amount of pixels.)

Anyways, awesome tool, I hope you can help or point me to some entry-level readings to understand how I can achieve this :))

**Comment by davidpagnon** [comment]
Thanks for your appreciation!

First of all, you don't need to read what follows, as I will change the behavior by the end of the week.
The origin of image coordinates usually starts from the top-left corner. The X-axis points to the right, and unintuitively, the Y-axis to the bottom. I did not have time to do something great so I just reverted the Y axis (pointing to the top from the top-left corner), and divided by 1000 to get lower values.
So if you want it in pure pixels, you need to multiply x coordinates by 1000, and y coordinates by -1000.

But I'm planning to add the possibility of calibrating the camera, either with a proper calibration method, or just by giving the height of one of the persons in the scene. 
Coordinates would be given in meters, Y would be pointing upwards, and the origin would be set either automatically (at the intersection between the line formed by the feet when in contact with the floor and the left side of the image), or manually (xy image coordinates). 
I'll ping you when it's done if you want!

P.S.: I must say that the current pose estimation algorithms do not work very well when the person is upside down, because there is little of it in the datasets they are trained on.

**Comment by Niels-Engedal** [comment]
Thank you so much! Please go ahead and ping me, when the update is live, until then I'll work with the multiplication of the coordinates :)

**Comment by davidpagnon** [comment]
Hi @Niels-Engedal and @JoanCharmant, 

I just pushed a release to obtain coordinates in meters rather than in pixels ([documentation in Readme](https://github.com/davidpagnon/Sports2D?tab=readme-ov-file#get-coordinates-in-meters-rather-than-in-pixels
)).

You just need to give it the height of a person, and its ID in case of multiple person detection.
The floor angle and the XY origin are detected automatically from gait, but they can also be a manual input if needed.

Feel free to test it and tell me if anything goes wrong!
To upgrade: `pip install sports2d -U`.

---

**PS:** I plan to make it accept more accurate calibration files, like in [Pose2Sim](https://github.com/perfanalytics/pose2sim). With the method based on person_height, motions in the frontal plane would not be very accurate (and distortions would not be taken into account).

**PS2:** I also added a few features, such as the possibility of loading a trc file instead of running pose estimation. In single-person mode, the person who is saved is now the one with the highest average confidence instead of just the first detected one. See [release notes](https://github.com/davidpagnon/Sports2D/releases/tag/v0.5.0)
